app/scripts/lib/app-start.js

```diff
@@ -596,6 +586,24 @@ function (
               }
             });
       });
+    },
+
+    _createMetrics: function (sampleRate, options) {
+      if (this._isAutomatedBrowser()) {
+        return new StorageMetrics(options);
+      } else if (this._isMetricsCollectionEnabled(sampleRate)) {
+        return new Metrics(options);
+      } else {
+        return new NullMetrics();
+      }
+    },
+
+    _isMetricsCollectionEnabled: function (sampleRate) {
+      return Math.random() <= sampleRate;
+    },
```

Comment:
This method is a bit weird because it is nondeterministic; it may give you different results if called multiple times, whereas other methods with the `_is` prefix are deterministic. Maybe it fits better as a standalone helper function so we avoid the confusion if someone tries to call it multiple times like they would with other `_is` methods.